numpy/_core/tests/test_defchararray.py

```diff
@@ -136,121 +137,130 @@ def fail():
 
 
 class TestWhitespace:
-    def setup_method(self):
-        self.A = np.array([['abc ', '123  '],
+    def test1(self):
+        A = np.array([['abc ', '123  '],
                            ['789 ', 'xyz ']]).view(np.char.chararray)
-        self.B = np.array([['abc', '123'],
+        B = np.array([['abc', '123'],
                            ['789', 'xyz']]).view(np.char.chararray)
+        assert_(np.all(A == B))
+        assert_(np.all(A >= B))
+        assert_(np.all(A <= B))
+        assert_(not np.any(A > B))
+        assert_(not np.any(A < B))
+        assert_(not np.any(A != B))
 
-    def test1(self):
-        assert_(np.all(self.A == self.B))
-        assert_(np.all(self.A >= self.B))
-        assert_(np.all(self.A <= self.B))
-        assert_(not np.any(self.A > self.B))
-        assert_(not np.any(self.A < self.B))
-        assert_(not np.any(self.A != self.B))
 
 class TestChar:
-    def setup_method(self):
-        self.A = np.array('abc1', dtype='c').view(np.char.chararray)
-
     def test_it(self):
-        assert_equal(self.A.shape, (4,))
-        assert_equal(self.A.upper()[:2].tobytes(), b'AB')
+        A = np.array('abc1', dtype='c').view(np.char.chararray)
+        assert_equal(A.shape, (4,))
+        assert_equal(A.upper()[:2].tobytes(), b'AB')
+
 
 class TestComparisons:
-    def setup_method(self):
-        self.A = np.array([['abc', 'abcc', '123'],
-                           ['789', 'abc', 'xyz']]).view(np.char.chararray)
-        self.B = np.array([['efg', 'efg', '123  '],
-                           ['051', 'efgg', 'tuv']]).view(np.char.chararray)
+    def _create_array_a(self):
+        return np.array([['abc', 'abcc', '123'],
+                            ['789', 'abc', 'xyz']]).view(np.char.chararray)
+
+    def _create_array_b(self):
+        return np.array([['efg', 'efg', '123  '],
+                            ['051', 'efgg', 'tuv']]).view(np.char.chararray)
```

Comment:
IMO this should be 

```
def _create_arrays(self):
    return (
        np.array([['abc', 'abcc', '123'], ['789', 'abc', 'xyz']]).view(np.char.chararray),
        np.array([['efg', 'efg', '123  '], ['051', 'efgg', 'tuv']]).view(np.char.chararray)
    )
```

Since in all cases this is what you want anyway, it's a little clearer just with one function call.